src/Symfony/Component/Validator/Tests/Constraints/EmailValidatorTest.php

```diff
@@ -86,4 +88,28 @@ public function getInvalidEmails()
             array('example@localhost'),
         );
     }
+
+    /**
+     * @dataProvider getHostnamesAndEmailAddresses
+     */
+    public function testHostnameIsProperlyParsed($hostnameAndEmail)
+    {
+        $method = new ReflectionMethod($this->validator, 'extractHostname');
+        $method->setAccessible(true);
```

Comment:
This test shouldn't use reflection. Just use the public API and create a test case that would fail without this change.